unified-scheduler-pool/src/lib.rs

```diff
@@ -0,0 +1,732 @@
+    result_with_timings: Mutex<Option<ResultWithTimings>>,
+}
+
+#[derive(Debug)]
+pub struct PooledSchedulerInner<S: SpawnableScheduler<TH>, TH: TaskHandler> {
+    id: SchedulerId,
+    pool: Arc<SchedulerPool<S, TH>>,
+}
+
+impl<TH: TaskHandler> PooledScheduler<TH> {
+    fn do_spawn(pool: Arc<SchedulerPool<Self, TH>>, initial_context: SchedulingContext) -> Self {
+        Self::from_inner(
+            PooledSchedulerInner::<Self, TH> {
+                id: pool.new_scheduler_id(),
+                pool,
+            },
+            initial_context,
+        )
+    }
+
+    fn do_wait_for_termination(&mut self) -> Option<ResultWithTimings> {
+        self.result_with_timings
+            .lock()
+            .expect("not poisoned")
+            .take()
+    }
+}
+
+pub trait SpawnableScheduler<TH: TaskHandler>: InstalledScheduler {
+    type Inner: Debug + Send + Sync;
+
+    fn into_inner(self) -> Self::Inner;
+
+    fn from_inner(inner: Self::Inner, context: SchedulingContext) -> Self;
+
+    fn spawn(pool: Arc<SchedulerPool<Self, TH>>, initial_context: SchedulingContext) -> Self
+    where
+        Self: Sized;
+}
+
+impl<TH: TaskHandler> SpawnableScheduler<TH> for PooledScheduler<TH> {
+    type Inner = PooledSchedulerInner<Self, TH>;
+
+    fn into_inner(self) -> Self::Inner {
+        self.inner
+    }
+
+    fn from_inner(inner: Self::Inner, context: SchedulingContext) -> Self {
+        Self {
+            inner,
+            context,
+            result_with_timings: Mutex::new(Some((Ok(()), ExecuteTimings::default()))),
+        }
+    }
+
+    fn spawn(pool: Arc<SchedulerPool<Self, TH>>, initial_context: SchedulingContext) -> Self {
+        Self::do_spawn(pool, initial_context)
+    }
+}
+
+impl<TH: TaskHandler> InstalledScheduler for PooledScheduler<TH> {
+    fn id(&self) -> SchedulerId {
+        self.inner.id
+    }
+
+    fn context(&self) -> &SchedulingContext {
+        &self.context
+    }
+
+    fn schedule_execution(&self, &(transaction, index): &(&SanitizedTransaction, usize)) {
+        let result_with_timings = &mut *self.result_with_timings.lock().expect("not poisoned");
+        let (result, timings) =
+            result_with_timings.get_or_insert_with(|| (Ok(()), ExecuteTimings::default()));
+
+        // ... so, we're NOT scheduling at all here; rather, just execute tx straight off. the
+        // inter-tx locking deps aren't needed to be resolved in the case of single-threaded FIFO
+        // like this.
+        if result.is_ok() {
+            TH::handle(
+                result,
+                timings,
+                self.context().bank(),
+                transaction,
+                index,
+                &self.inner.pool.handler_context,
+            );
+        }
+    }
+
+    fn wait_for_termination(
+        mut self: Box<Self>,
+        _is_dropped: bool,
+    ) -> (Box<dyn UninstalledScheduler>, ResultWithTimings) {
+        let result_with_timings = self.do_wait_for_termination().unwrap();
```

Comment:
oops. note that this `.unwrap()` will be replaced soon with upcoming pr...

Reply:
well, i did this in this pr: https://github.com/solana-labs/solana/pull/34300/commits/ef4874dc5641913297a0558e074f938f27f64b7c